presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTaskFactory.java

```diff
@@ -165,4 +183,73 @@ public RemoteTask createRemoteTask(Session session,
                 stats,
                 isBinaryTransportEnabled);
     }
+
+    @Override
+    public ListenableFuture<?> removeRemoteSource(TaskId taskId, TaskId remoteSourceTaskId)
+    {
+        URI remoteSourceUri = uriBuilderFrom(locationFactory.createLocalTaskLocation(taskId))
+                .appendPath("remote-source")
+                .appendPath(remoteSourceTaskId.toString())
+                .build();
+
+        Request request = prepareDelete()
+                .setUri(remoteSourceUri)
+                .build();
+        RequestErrorTracker errorTracker = new RequestErrorTracker(
+                taskId,
+                remoteSourceUri,
+                maxErrorDuration,
+                errorScheduledExecutor,
+                "Remove exchange remote source");
+
+        SettableFuture<?> future = SettableFuture.create();
+        removeRemoteSourceHelper(errorTracker, request, future);
+        return future;
+    }
+
+    private void removeRemoteSourceHelper(RequestErrorTracker errorTracker, Request request, SettableFuture<?> future)
+    {
+        errorTracker.startRequest();
+
+        FutureCallback<StatusResponse> callback = new FutureCallback<StatusResponse>() {
+            @Override
+            public void onSuccess(@Nullable StatusResponse response)
+            {
+                if (response == null) {
+                    throw new RuntimeException("Request failed with null response");
+                }
+                if (response.getStatusCode() != HttpStatus.OK.code()) {
```

Comment:
I think you can static import `HttpStatus.OK` here.

Reply:
@nezihyigitbasi : I found it tricky to decide for very short things like `OK`. I think here we can do static import since we know it's an `HttpStatus.OK` from the context.